src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -524,6 +524,27 @@ public override BoundNode VisitYieldReturnStatement(BoundYieldReturnStatement no
             return UpdateStatement(builder, node.Update(expression));
         }
 
+        public override BoundNode VisitCatchBlock(BoundCatchBlock node)
+        {
+            BoundExpression exceptionSourceOpt = (BoundExpression)this.Visit(node.ExceptionSourceOpt);
+            var locals = node.Locals;
+
+            var exceptionFilterPrologueOpt = node.ExceptionFilterPrologueOpt;
+            Debug.Assert(exceptionFilterPrologueOpt is null); // it is introduced by this pass
+            BoundSpillSequenceBuilder builder = null;
+            var exceptionFilterOpt = VisitExpression(ref builder, node.ExceptionFilterOpt);
+            if (builder is { })
+            {
+                Debug.Assert(builder.Value is null);
+                locals = locals.AddRange(builder.GetLocals());
```

Comment:
This will put the locals from the `catch` before the filter even though filter executes first. Does that have any impact on the emitted code? Don't believe it does.

Reply:
These locals are all temps created by rewriting code so that statements are at the top level. It doesn't matter much where they go as long as they are surrounding the uses.

There are a number of places in the code that assumes that a catch variable, if there is one, is first in the list of variables. This ensures that remains correct.

---
In reply to: [456699808](https://github.com/dotnet/roslyn/pull/46054#discussion_r456699808) [](ancestors = 456699808)